WooCommerce/WooCommerceTests/ViewRelated/Dashboard/StoreStats/StoreStatsChartViewModelTests.swift

```diff
@@ -150,6 +150,22 @@ final class StoreStatsChartViewModelTests: XCTestCase {
         XCTAssertEqual(viewModel.yAxisStride, 100)
     }
 
+    func test_yAxisStride_can_handle_negative_revenue_values() {
+        // Given
+        let currentDate = Date()
+        let intervals = [
+            StoreStatsChartData(date: currentDate, revenue: 100),
+            StoreStatsChartData(date: currentDate.adding(days: 1)!, revenue: -10),
+            StoreStatsChartData(date: currentDate.adding(days: 2)!, revenue: 130),
+        ]
+
+        // When
+        let viewModel = StoreStatsChartViewModel(intervals: intervals, timeRange: .thisWeek)
+
+        // Then
+        XCTAssertEqual(viewModel.yAxisStride, 70)
+    }
```

Comment:
@selanthiraiyan this test could be improved a bit:
- The crash was due to really small negative revenues.
- The test should check that `yAxisStride` is larger than 0 because that was what crashed the Charts library.

Reply:
Thanks, @itsmeichigo!

> The crash was due to really small negative revenues.

When I tested, the store had revenues in the range of -39, -88, -19.5. Do you think we should use decimal values? 

> The test should check that yAxisStride is larger than 0 because that was what crashed the Charts library.

Good point. I will update this in one of my future PRs.

Reply:
For the app to crash, the smallest negative revenue has to have a larger absolute value than the largest positive value for the `yAxisStride` to be negative. So the test case above should be:

```suggestion
    func test_yAxisStride_can_handle_negative_revenue_values() {
        // Given
        let currentDate = Date()
        let intervals = [
            StoreStatsChartData(date: currentDate, revenue: 100),
            StoreStatsChartData(date: currentDate.adding(days: 1)!, revenue: 10),
            StoreStatsChartData(date: currentDate.adding(days: 2)!, revenue: -130),
        ]

        // When
        let viewModel = StoreStatsChartViewModel(intervals: intervals, timeRange: .thisWeek)

        // Then
        XCTAssertTrue(viewModel.yAxisStride > 0)
    }
```